src/classLibrary/ParsingStrategies/InlineFlagResolver.ts

```diff
@@ -0,0 +1,174 @@
+		})
+		return allPossibleInverseInlineFlags;
+	}
+	
+	ResolveFiles(cyan: CyanSafe, virtualFiles: VirtualFileSystemInstance[]): VirtualFileSystemInstance[] {
+		const flagsMap: Map<string, boolean> = this.util.FlattenObject(cyan.flags).MapValue((boolString: string) => boolString == "true");
+		const allPossibleFlagsMap: Map<string[], boolean> = flagsMap.MapKey((flag: string) => this.ModifyFlagWithAllSyntax(flag, cyan.syntax));
+        const allPossibleInverseFlagsMap: Map<string[], boolean> = flagsMap.MapKey((flag: string) => this.ModifyInverseFlagWithAllSyntax(flag, cyan.syntax));
+
+		return virtualFiles.Where((virtualFile: VirtualFileSystemInstance) => {
+            let shouldStay: boolean = true;
+            allPossibleFlagsMap
+                .forEach((flagVal: boolean, flagWithAllSyntaxes: string[]) => {
+                    shouldStay = shouldStay && this.ShouldKeepFileWithInlineFlag(flagWithAllSyntaxes, flagVal, virtualFile);
+                })
+            allPossibleInverseFlagsMap
+                .forEach((flagVal: boolean, flagWithAllSyntaxes: string[]) => {
+                    shouldStay = shouldStay && this.ShouldKeepFileWithInlineFlag(flagWithAllSyntaxes, !flagVal, virtualFile);
+                })
+            return shouldStay;
+        }).Each((virtualFile: VirtualFileSystemInstance) => {
+            allPossibleFlagsMap.Keys().concat(allPossibleInverseFlagsMap.Keys())
+                .forEach((key: string[]) => {
+                    this.RemoveFlagFromMetadataDestinationPath(key, virtualFile)
+                });    
+        });
+	};
+
+    ShouldKeepFileWithInlineFlag(flagSignatures: string[], flagValue: boolean, virtualFile: VirtualFileSystemInstance): boolean {
+        return VirtualFileSystemInstance.match(virtualFile, {
+            File: (file: FileSystemInstance) => {
+                const checkTarget: string = file.metadata.relativePath;
+                return this.ShouldKeepStringWithInlineFlag(flagSignatures, flagValue, checkTarget);
+            },
+            Folder: (folder: DirectorySystemInstance) => {
+                const checkTarget: string = folder.metadata.relativePath;
+                return this.ShouldKeepStringWithInlineFlag(flagSignatures, flagValue, checkTarget);     
+            },
+            default: () => false,
+        });
+	}
+
+    ShouldKeepStringWithInlineFlag(flagSignatures: string[], flagValue: boolean, checkTarget: string): boolean {
+        return flagSignatures.Where((signature: string) => !checkTarget.includes(signature) || flagValue).length === flagSignatures.length;   
+    }
+
+    RemoveFlagFromMetadataDestinationPath(signatures: string[], virtualFile: VirtualFileSystemInstance) {
+        VirtualFileSystemInstance.match(virtualFile, {
+            File: (file: FileSystemInstance) => {
+                file.metadata.destinationAbsolutePath = file.metadata.destinationAbsolutePath.Without(signatures.Map((s: string) => s));
```

Comment:
Can be simplified to `file.metadata.destinationAbsolutePath.Without(signatures);`